File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    
    int n = nums.size();
    n /= 2;
    
    sort(nums.begin(), nums.end());
    nums.erase(unique(nums.begin(), nums.end()), nums.end());
    
    if(nums.size() < n) answer = nums.size();
    else answer = n;
    
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    hash = {}
    
    for i in nums:
        hash[i] = 1                 
        
    if len(nums) // 2 <= len(hash): 
        return len(nums) // 2         
    
    return len(hash)       
    
    return answer